File: app/src/main/java/com/github/ykiselev/playground/init/GlfwBootstrap.java
package com.github.ykiselev.playground.init;

import static org.lwjgl.glfw.GLFW.glfwInit;
import static org.lwjgl.glfw.GLFW.glfwTerminate;

public final class GlfwBootstrap implements AutoCloseable {

    public GlfwBootstrap() {
        glfwInit();
    }

    @Override
    public void close() {
        glfwTerminate();
    }
}

File: app/src/main/kotlin/com/github/ykiselev/playground/init/GlfwBootstrap.kt
package com.github.ykiselev.playground.init

import org.lwjgl.glfw.GLFW

class GlfwBootstrap : AutoCloseable {
    init {
        GLFW.glfwInit()
    }

    override fun close() {
        GLFW.glfwTerminate()
    }
}
